### ISSUE TITLE
[Deque Re-Analysis] Submission "critical" accessibility issues

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1316]
**Describe the bug**
In Sept 2021, Deque did a reanalysis of the issues described in #1171 which were fixed in #1252.  

They found the following 9 accessibility issues were either "Not Fixed" or only "Partially Fixed":

### Partially Fixed Issues
1. (Issue ID: 469979) "Keyboard Navigation" - Arrows above/below date field (to change year/month/day) are inaccessible to keyboard navigation (you cannot tab to them).  (_Mostly works now_)
    * Deque's re-test notes: "The arrow icon buttons to increment and decrement values are receiving keyboard focus. However, they lack focus indicator and sighted keyboard users will not be able to identify the elements in focus."
2. (Issue ID 470065) Section headers open/close icons (up/down arrows) have incorrect names/labels.
    * Deque's re-test notes: "The expand/collapse arrow icon buttons are provided with `aria-label="Close section"`. More clear name like "Expand Describe Section", "Collapse Describe Section" is recommended."

### Not Fixed Issues
While Deque flagged these as "Not Fixed", some behave slightly better, but the overall accessibility problem is not fixed.
1. (Issue ID: 469700) "Button does not have a name or role" - "browse" link in the file drag & drop component (`<input type="file">`) has no "role". 
    * Deque's re-test notes: "The Browse element is not being exposed as a button to NVDA. Please provide role="button" to the `<label>` element instead of on the `<input>` element."
2.  (Issue ID 469984) Group of form controls is missing "group" label for the "Date of Issue" field.
    *  Deque's re-test notes: "The three fields of day, month and year are not grouped as "Date of Issue *". Use `<fieldset>` element to group the elements and provide group name ("Date of Issue *") using `<legend>` element."
3. (Issue ID 469993) Two comboboxes are missing roles/attributes. Specifically "dc.type" and "dc.language.iso" fields.
    * Deque's re-test notes: "The comboboxes Type and Language do not have sufficient ARIA attributes. They are current being announced as edit boxes."
4. (Issue 470001) No focus indicator exists on File buttons (Download, Edit, Delete...and Save/Cancel after clicking Edit).
    * Deque's re-test notes: "Focus indicator is missing for the icon button. `outline: 0;` is provided to these elements which is removing the focus indicator."
5. (Issue ID 470010) Form field with error not identified. Occurs on the "Grant Access" fields in the Bitstream Embargo section. When an error occurs it is not linked to the field.
    * Deque's re-test notes: "Either include the form field name in the error message like `"Grant access from date is required"`. Or associate the error message element with its form field using `aria-describedby`."
6. (Issue ID: 470011) Bitstream embargo fields have a keyboard trap (Upload a file -> Edit -> "Access Condition" to "embargo" or "lease" -> Attempt to tab past the "Grant access..." field). It's not possible to tab past them until you fill them out (Keyboard focus is trapped within the field). 
    * Deque's re-test notes: "Keyboard trap is still present at Grant access from and until fields in errored state."
7. (Issue ID 470062) Section status icons are missing alt text. When a submission section has a green checkmark, it has all required fields filled. Otherwise it shows an orange exclamation mark. Both those status icons are missing alt text.
    * Deque's re-test notes: "The informative icons are provided with aria-hidden="true" incorrectly."

The above issues should be fixed

**Comment by tdonohue** [comment]
@atarix83 : I know you have a lot on your plate right now. But this ticket is the result of a re-test by Deque of the accessibility issues you tried to fix in #1252.  While a lot of the fixes in that old PR were successful, there were 9 listed here that are not quite fixed.  Many of these could be quick to fix if you or someone on your team finds time before the 7.1 release.
